pkg/devicerepository/store/remote/schema.go

```diff
@@ -0,0 +1,460 @@
+	VendorProfileID uint32 `yaml:"vendorProfileID"`
+	SupportsClassB  bool   `yaml:"supportsClassB"`
+	ClassBTimeout   uint32 `yaml:"classBTimeout"`
+	PingSlotPeriod  uint32 `yaml:"pingSlotPeriod"`
+
+	PingSlotDataRateIndex     ttnpb.DataRateIndex `yaml:"pingSlotDataRateIndex"`
+	PingSlotFrequency         float64             `yaml:"pingSlotFrequency"`
+	SupportsClassC            bool                `yaml:"supportsClassC"`
+	ClassCTimeout             uint32              `yaml:"classCTimeout"`
+	MACVersion                ttnpb.MACVersion    `yaml:"macVersion"`
+	RegionalParametersVersion string              `yaml:"regionalParametersVersion"`
+	SupportsJoin              bool                `yaml:"supportsJoin"`
+	Rx1Delay                  ttnpb.RxDelay       `yaml:"rx1Delay"`
+	Rx1DataRateOffset         uint32              `yaml:"rx1DataRateOffset"`
+	Rx2DataRateIndex          ttnpb.DataRateIndex `yaml:"rx2DataRateIndex"`
+	Rx2Frequency              float64             `yaml:"rx2Frequency"`
+	FactoryPresetFrequencies  []float64           `yaml:"factoryPresetFrequencies"`
+	MaxEIRP                   float32             `yaml:"maxEIRP"`
+	MaxDutyCycle              float64             `yaml:"maxDutyCycle"`
+	Supports32BitFCnt         bool                `yaml:"supports32bitFCnt"`
+}
+
+var errRegionalParametersVersion = errors.DefineNotFound("regional_parameters_version", "unknown Regional Parameters version `{phy_version}`")
+
+// ToTemplatePB returns a ttnpb.EndDeviceTemplate from an end device profile.
+func (p EndDeviceProfile) ToTemplatePB(ids *ttnpb.EndDeviceVersionIdentifiers, info *ttnpb.EndDeviceModel_FirmwareVersion_Profile) (*ttnpb.EndDeviceTemplate, error) {
+	phyVersion, ok := regionalParametersToPB[p.RegionalParametersVersion]
+	if !ok {
+		return nil, errRegionalParametersVersion.WithAttributes("phy_version", p.RegionalParametersVersion)
+	}
+
+	paths := []string{
+		"version_ids",
+		"supports_join",
+		"supports_class_b",
+		"supports_class_c",
+		"lorawan_version",
+		"lorawan_phy_version",
+	}
+	dev := ttnpb.EndDevice{
+		VersionIDs:        ids,
+		SupportsJoin:      p.SupportsJoin,
+		SupportsClassB:    p.SupportsClassB,
+		SupportsClassC:    p.SupportsClassC,
+		LoRaWANVersion:    p.MACVersion,
+		LoRaWANPHYVersion: phyVersion,
+	}
+
+	if info.CodecID != "" {
+		dev.Formatters = &ttnpb.MessagePayloadFormatters{
+			DownFormatter: ttnpb.PayloadFormatter_FORMATTER_REPOSITORY,
+			UpFormatter:   ttnpb.PayloadFormatter_FORMATTER_REPOSITORY,
+		}
+		paths = append(paths, "formatters")
+	}
+
+	dev.MACSettings = &ttnpb.MACSettings{}
+	if p.ClassBTimeout > 0 {
+		t := time.Duration(p.ClassBTimeout) * time.Second
+		dev.MACSettings.ClassBTimeout = &t
+		paths = append(paths, "mac_settings.class_b_timeout")
+	}
+	if p.ClassCTimeout > 0 {
+		t := time.Duration(p.ClassCTimeout) * time.Second
+		dev.MACSettings.ClassCTimeout = &t
+		paths = append(paths, "mac_settings.class_c_timeout")
+	}
+	if p.PingSlotDataRateIndex > 0 {
```

Comment:
`0` is a valid value, also note that this and all other `ping.*` parameters should always be set when `SupportsClassB` is `true` according to LoRaWAN Backend interfaces

Reply:
That validation is also enforced by the Device Repository so you can rely on the presence of those values when `SupportsClassB` is set